wordpress-mobile/WordPress-iOS
# MediaLibrary: MediaExporter protocol, extension, errors, structs and tests.

Continuing from https://github.com/wordpress-mobile/WordPress-iOS/pull/7116, this PR introduces the base declaration of the `MediaExporter` protocol, it's default extension and supporting declarations.

Note: Exporters are what handle the grunt work of taking whatever Media is selected from whatever source, and processing it for upload. For the most part, I've followed all the hard work already put in to the existing implementation, then refined and consolidated.

- The `MediaExporter` protocol defines what's required of any conforming exporters, such as image exporters, URL exporters, asset exporters, etc.
- The `MediaExporter` `extension` implements a couple default helper methods shared among exporters.
- The `MediaExportError` protocol defines the required conformance for any errors generated by an exporter, which is also handy as a value type.
- `MediaExportSystemError` is a generic `MediaExportError` used for wrapping general system errors encountered by an exporter.
- The `MediaExport` protocol and conforming structs defined what is required of an export that results with any of those particular kind of exports, such as an image or video.

To test:
1. Make sure the code compiles.
2. Run the tests for `MediaExporterTests.swift`

Needs review: @frosty and @SergioEstevao round two :bowtie:
